### PR TITLE
Convert integration tests to Cypress

### PR DESCRIPTION
**Pull request by nemanjaglumac**
### Status
READY

### What does this PR accomplish?
- Converts remaining integration tests to Cypress test
- Removes all integration tests and their related files
    - There is one more related file left `frontend/test/__support__/integration.js` which is used by some unit test files. Once we remove them, it will be safe to remove it as well.
- Removes `fe-tests-integration` from CI

#### Context:
- Main list at https://github.com/metabase/metabase/issues/13657

### Additional notes and qeustions
- I have removed `AuditTable.integ.spec.jsx` in 1aa9aa225fc753cb8ef871a8154c75a00159c905. Although the file name suggests it should test audits, it was basically testing that there is pagination present when there are more than X rows.

There was this mock POST api call to generate the needed rows:
```javascript
mock.post("/api/dataset", (req, res) =>
  res.json({
    data: { rows: [[0]], cols: [{ name: "x" }] },
    row_count: 100,
    status: "completed",
  }),
);
```
My question is do we really need to migrate this to Cypress? If there is value in doing that, I'll follow up with another PR in order not to block this merge.

### CODE REVIEW

File: docs/developers-guide.md [hunk 1]
```diff
@@ -120,7 +121,7 @@ All frontend tests are located in `frontend/test` directory. Run all frontend te
 yarn test
 ```
 
-which will run unit, integration and Cypress end-to-end tests in sequence.
+which will run unit, timezones and Cypress end-to-end tests in sequence.
```
Comment (nemanjaglumac) [inline docs/developers-guide.md:124]: Maybe I should've omitted `timezones`? After all, they are a form of unit tests if I'm not mistaken.

File: frontend/test/metabase/scenarios/setup/user_settings.cy.spec.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    cy.findByText("Update").click();
+    cy.findByDisplayValue("John");
+
+    // It is hard and unreliable to assert that something didn't happen in Cypress
+    // This solution was the only one that worked out of all others proposed in this SO topic: https://stackoverflow.com/a/59302542/8815185
+    cy.get("@membership").then(() => {
+      expect(
+        cy.state("requests").filter(r => r.alias === "membership"),
+      ).to.have.length(0);
+    });
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/setup/user_settings.cy.spec.js:39]: To test whether this really catches an XHR, I intentionally changed XHR to the one that is present. The test fails as expected:
image

File: frontend/test/metabase/scenarios/setup/user_settings.cy.spec.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      .clear()
+      .type("John");
+    cy.findByText("Update").click();
+    cy.findByDisplayValue("John");
+
+    // It is hard and unreliable to assert that something didn't happen in Cypress
+    // This solution was the only one that worked out of all others proposed in this SO topic: https://stackoverflow.com/a/59302542/8815185
+    cy.get("@membership").then(() => {
+      expect(requestsCount("membership")).to.have.length(0);
+    });
```
Comment (nemanjaglumac) [inline frontend/test/metabase/scenarios/setup/user_settings.cy.spec.js:36]: To test whether this really catches an XHR, I intentionally changed XHR **locally** to the one that is present. The test failed as expected:
image

File: docs/developers-guide.md (hunk 1 above)
Comment (ariya) [reply docs/developers-guide.md:124]: Yes, it is a form of unit tests (although I'm not sure why there is a separate Jest config for that, though)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,10 +1,11 @@
...
+
 - [ ] If you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.
+
 ### Tests
--  [ ] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)
--  [ ] If there are changes to the backend codebase, run the backend tests with `lein test && lein lint && ./bin/reflection-linter`
 
--  [ ] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
-(unless it's a tiny documentation change).
+- [ ] Run the frontend and Cypress end-to-end tests with `yarn lint && yarn flow && yarn test`)
```
Comment (ariya) [inline .github/PULL_REQUEST_TEMPLATE.md:7]: An opportunity to fix this: `yarn lint` already runs Flow (among others, such as ESLint, Prettier, etc).

File: docs/developers-guide.md (hunk 1 above)
Comment (paulrosenzweig) [reply docs/developers-guide.md:124]: We run those with different pairs of client/reporting timezones. In the context of unit tests, the "client" timezone is node's timezone.

There have been bugs caused by the browser being east or west of the MB reporting timezone.

[Here's where we run those.](https://github.com/metabase/metabase/blob/master/frontend/test/__runner__/run_timezone_tests)

Comment (nemanjaglumac) [reply docs/developers-guide.md:124]: So, the final verdict is...? :)
Leave that piece of text or remove it? Not that it makes much difference, but I'm asking simply for the "correctness" sake.

Comment (paulrosenzweig) [reply docs/developers-guide.md:124]: Ah sorry I didn't comment on that. I might drop it, but either way is fine.

**Review by ariya — APPROVED** [review]
Beside that nitpicky issue, it's a solid refactoring :+1:

File: docs/developers-guide.md (hunk 1 above)
Comment (nemanjaglumac) [reply docs/developers-guide.md]: Dropped, done :)
Thanks

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 4 above)
Comment (nemanjaglumac) [reply .github/PULL_REQUEST_TEMPLATE.md]: Fixed it. Thank you for valuable suggestion.

**Pull request merged**